### PR TITLE
Partial implementation of HTTP-network Fetch step

### PR DESCRIPTION
**Pull request by nikkisquared**
I've been working on making a partial implementation of HTTP-network Fetch[1] so that I can begin testing the entire Fetch protocol as implemented in Servo. The empty steps are intentionally so, including the incomplete Step 5, which I started without realizing I don't need to do basic testing.

[1] https://fetch.spec.whatwg.org/#http-network-fetch

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9153)

### CODE REVIEW

File: components/net/fetch/request.rs [hunk 1]
```diff
@@ -4,16 +4,20 @@
 
 use fetch::cors_cache::{BasicCORSCache, CORSCache, CacheRequestDetails};
 use fetch::response::ResponseMethods;
+use http_loader::{NetworkHttpRequestFactory, WrappedHttpResponse};
+use http_loader::{create_http_connector, obtain_response};
```
Comment (KiChjang) [inline components/net/fetch/request.rs]: I'm not sure whether it's a good idea for the fetch infrastructure to depend upon `http_loader`.

**Comment by frewsxcv** [comment]
*[components/net_traits/lib.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9153#-K7EopAKuIwz3h6WMPhn:-K7EopAKuIwz3h6WMPho:1567228796) ([raw file](https://github.com/servo/servo/blob/4440d1958b7fd0bf9bfa9419c3da0b46b06902a9/components/net_traits/lib.rs#L84)):*
A couple things:

1. Considering you're using markdown, it might be helpful to use `///` so that it gets rendered in the documentation
2. You only need one set of `[` and `]` for links in markdown

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9153)*

**Comment by frewsxcv** [comment]
*[components/net/fetch/request.rs, line 784 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9153#-K7EpIBSKbaay7dxDGMI:-K7EpIBSKbaay7dxDGMJ:610402904) ([raw file](https://github.com/servo/servo/blob/4440d1958b7fd0bf9bfa9419c3da0b46b06902a9/components/net/fetch/request.rs#L784)):*
This could be:

```
if let Some(encoding) = response.headers.get::<ContentEncoding>() {
    ...
}
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9153)*

**Comment by jdm** [comment]
Nice work! With @frewsxcv's comments addressed, the first commit removed and the remaining ones squashed together, and the result rebased against master, I'd be happy to merge these changes.

**Pull request merged**